### PR TITLE
Define new index route and fixed login issues

### PR DESCRIPTION
**Pull request by pratanothayanon97**
There are some changes made as following
- Create new homepage in main/index.html.erb instead of user/show.html.erb
- Initialize point and privileges id to zero when creating new user
- Fixed all users CRUD
- Fixed integration testing

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -3,10 +3,10 @@ class ApplicationController < ActionController::Base
...
     def require_login
-    redirect_to new_session_path unless session.include? :user_id
+        redirect_to new_session_path unless session.include? :user_id
     end
 
     def current_user
-    @current_user ||= User.find(session[:user_id]) if session[:user_id]
+        @current_user ||= User.find(session[:user_id]) if session[:user_id]
     end
 end
```
Comment (amaredia3) [inline app/controllers/application_controller.rb:12]: Thanks for updating the formatting!

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -9,13 +9,17 @@ def create
...
       session[:user_id] = @user.id
-      redirect_to @user
+      # if email and password matches, redirect to homepage
+      redirect_to :root
     else
-      flash[:notice] = "Login is invalid!"
+      flash[:notice] = "Login failed! Please check your email or password."
       redirect_to new_session_path
     end
   end
```
Comment (amaredia3) [inline app/controllers/sessions_controller.rb:19]: Good idea to provide more descriptive error messages to user.

File: coverage/.last_run.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+  "result": {
+    "line": 91.81
+  }
+}
```
Comment (amaredia3) [inline coverage/.last_run.json:5]: Great code coverage!

**Review by amaredia3 — APPROVED** [review]

File: app/controllers/sessions_controller.rb (hunk 2 above)
Comment (caleb-terry17) [reply app/controllers/sessions_controller.rb:19]: thank you for adding the redirection to homepage after a user logs in!

**Review by caleb-terry17 — APPROVED** [review]
follows all coding conventions, and does the following: 

- passes all integration and unit tests
- for user story "Point Value System" so that admins can assign point values to events to be awarded to users for attending events
- made login failure notice more descriptive
- made formatting changes 
- added controller for homepage 
- added redirection to homepage after user logs in
- added views for users

**Pull request merged**